cmd/gubernator/main.go

```diff
@@ -87,16 +87,27 @@ func main() {
 		checkErr(err, "while querying kubernetes API")
 	} else {
 		// Register ourselves with other peers via ETCD
-		etcdClient, err := etcdutil.NewClient(&conf.EtcdConf)
-		checkErr(err, "while connecting to etcd")
-
-		pool, err = gubernator.NewEtcdPool(gubernator.EtcdPoolConfig{
-			AdvertiseAddress: conf.EtcdAdvertiseAddress,
-			OnUpdate:         guber.SetPeers,
-			Client:           etcdClient,
-			BaseKey:          conf.EtcdKeyPrefix,
+		// etcdClient, err := etcdutil.NewClient(&conf.EtcdConf)
+		// checkErr(err, "while connecting to etcd")
+
+		// pool, err = gubernator.NewEtcdPool(gubernator.EtcdPoolConfig{
+		// 	AdvertiseAddress: conf.EtcdAdvertiseAddress,
+		// 	OnUpdate:         guber.SetPeers,
+		// 	Client:           etcdClient,
+		// 	BaseKey:          conf.EtcdKeyPrefix,
+		// })
+		// checkErr(err, "while registering with ETCD pool")
+
+		// Register peer on memberlist
+		pool, err = gubernator.NewMemberlistPool(gubernator.MemberlistPoolConfig{
```

Comment:
How confident are we that MemberList works better than etcd? In any case we should make etcd an optional thing if k8s or etcd not selected in the config then we should fall back to memberlist. Personally I'm super excited about memberlist.